README.md

```diff
@@ -29,9 +29,9 @@ Rainbow uses the same Go code as the HTTP gateway in Kubo, but is fully speciali
 
   * Rainbow acts as DHT and Bitswap client only. Rainbow is not a server for the network.
   * Rainbow does not pin, or permanently store any content. It is just meant
-    to act as gateway to content present in the network. GC strategy 
+    to act as gateway to content present in the network. GC strategy
   * Rainbow settings are optimized for production deployments and streamlined
-    for specific choices (badger datastore, writethrough uncached blockstore
+    for specific choices (flatfs datastore, writethrough uncached blockstore
```

Comment:
My 2c is that I think we're still figuring out what datastore is going to work best here. While I don't think we currently need a ton of configurability on the datastores here having a few seems like a good idea as we deploy and figure this out.

A compromise might be to have add a CLI flag for `datastore` that is one of FlatFS or Badger (or some other fixed presets that people are finding useful like pebble) that gets passed into `Config` and then just have a switch statement for the datastores.

As rainbow gets some time for people to run it in a few environments we can revisit what works here. However, I'd take it as an assumption that the rainbow datastore doesn't need to persist and therefore can break whenever (e.g. changing the FlatFS function to have more/less folder sharding, upgrading Badger versions, ...).

---

FWIW to me FlatFS is kind of the "dumb but works" option. However, people have been known to run into IOPS, file descriptor, etc. issues at high read volumes. Having it be the default until something that is low maintenance and works better is found sounds good to me.

Reply:
The switch statement idea isn't a bad one. On the other hand, it appears that a few folks are already forking and building rainbow with non-badger datastores, so it does seem like badger needs to go regardless of what the end result is.

At this point, I don't think I've ever seen badger GC actually work.